tests/wsl/install_wsl.pm

```diff
@@ -0,0 +1,128 @@
+            timeout => 120
+        );
+    } else {
+        # In older versions, the certificate is included in the .appx file
+        # a) Open the image file in Explorer
+        $self->run_in_powershell(cmd => q{ii C:\\});
+        assert_and_click 'wsl-appx-file', timeout => 60, button => 'right';
+        wait_still_screen stilltime => 3, timeout => 10;
+        # b) Reach certificate installation
+        assert_and_click 'open-file-properties';
+        assert_and_click 'digital-signatures', timeout => 60;
+        assert_and_click 'build-service-cert';
+        assert_and_click 'cert-details';
+        assert_and_click 'view-certificate-details';
+        assert_and_click 'install-certificate', timeout => 60;
+        wait_still_screen stilltime => 3, timeout => 10;
+        assert_and_click 'install-certificate-to-local-machine';
+        wait_screen_change(sub { send_key 'ret' }, 10);
+        assert_screen 'user-acount-ctl-allow-make-changes', 20;
+        assert_and_click 'user-acount-ctl-yes';
+        wait_still_screen stilltime => 3, timeout => 10;
+        assert_and_click 'select-custom-store', timeout => 120;
+        wait_still_screen stilltime => 1, timeout => 5;
+        assert_and_click 'browse';
+        assert_and_click 'select-store-trusted-roots';
+        # confirm selected certificate store and then
+        # hit next
+        wait_screen_change(sub { send_key 'ret' }, 10) for (1 .. 2);
+        assert_screen 'completing-cert-import';
+        assert_and_click 'finish-button-in-win10';
+        assert_screen 'successful-certificate-import';
+        # close all opened windows, including powershell
+        send_key_until_needlematch 'powershell-ready-prompt', 'alt-f4', 26, 2;
+    }
+}
+
+sub run {
+    my ($self) = @_;
+    assert_screen 'windows-desktop';
+
+    $self->open_powershell_as_admin;
+    # Set the version for WSL1
+    $self->run_in_powershell(
+        cmd => 'wsl --set-default-version 1',
+        timeout => 30
+    ) unless (get_var('WSL2'));
+
+    my $WSL_version = '';
+    if (is_sle('<=15-sp4')) {
+        $WSL_version = "SUSE-Linux-Enterprise-Server-" . get_required_var("VERSION");
+    } elsif (is_sle('>=15-sp5')) {
+        $WSL_version = "SUSE-Linux-Enterprise-" . get_required_var("VERSION");
+    } elsif (is_leap) {
+        $WSL_version = "openSUSE-Leap-" . get_required_var("VERSION");
+    } else {
+        $WSL_version = "openSUSE-Tumbleweed";
+    }
+    my $install_from = get_required_var('WSL_INSTALL_FROM');
+    if ($install_from eq 'msstore') {
+
+        # Install required SUSE distro from the MS Store
+        $self->run_in_powershell(
+            cmd => "wsl --install --distribution $WSL_version",
+            code => sub {
+                assert_screen("yast2-wsl-firstboot-welcome", timeout => 300);
+            }
+        );
+    } elsif ($install_from eq 'build') {
```

Comment:
I think this should be the default state as primary we are pulling the images from build service

Reply:
Done